Issue-842: Fix deprecation warnings during build time (external dependencies)

v3rtumnus opened the pull request:
Fixes #842 

This is only the beginning but I also wanted to get feedback from your CI server.

v3rtumnus commented:
@carlspring This pull now contains fixes to all deprecations in regards to external libraries, with one exception: The use of `StandardPasswordEncoder` in the class `CacheManagerAuthenticationCache`, since I was not sure if you really want to replace it with one of the other encoders. I will add a list of internal deprecations in issue #842

carlspring commented:
Thanks for looking into it! So far, I think it looks good!

@sbespalov / @fuss86 : Any thoughts regarding the `StandardPasswordEncoder` and `CacheManagerAuthenticationCache` question?

@v3rtumnus : The list would be quite helpful, thanks!

carlspring commented:
@v3rtumnus : Is this ready for merging?

v3rtumnus commented:
It currently contains all fixes for external deprecation warnings (except the `StandardPasswordEncoder)`. If you want to do the internal ones (hope to provide the list in one of the next days) in a separate PR, then it is ready from my side.

v3rtumnus commented:
Here is a list of internal deprecations:

```
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/storage/validation/resource/ArtifactOperationsValidator.java: Some input files use or override a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/MockedRemoteRepositoriesHeartbeatConfig.java: Some input files use or override a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/storage/indexing/RepositoryIndexer.java: Some input files use or override a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/storage/validation/groupid/MavenGroupIdLowercaseValidatorTest.java: Some input files use or override a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/NugetLayoutProvider.java: /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/NugetLayoutProvider.java uses or overrides a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/config/NugetLayoutProviderTestConfig.java: Some input files use or override a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/NpmLayoutProvider.java: /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/NpmLayoutProvider.java uses or overrides a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-raw-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/RawLayoutProvider.java: /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-raw-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/RawLayoutProvider.java uses or overrides a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-raw-layout-provider/src/test/java/org/carlspring/strongbox/providers/layout/RawLayoutProviderTest.java: /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-raw-layout-provider/src/test/java/org/carlspring/strongbox/providers/layout/RawLayoutProviderTest.java uses or overrides a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-p2-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/P2LayoutProvider.java: /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-p2-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/P2LayoutProvider.java uses or overrides a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-web-core/src/main/java/org/carlspring/strongbox/web/HeaderMappingFilter.java: /home/michael/Entwicklung/workspace_test/strongbox/strongbox-web-core/src/main/java/org/carlspring/strongbox/web/HeaderMappingFilter.java uses or overrides a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-web-core/src/test/java/org/carlspring/strongbox/config/IntegrationTest.java: /home/michael/Entwicklung/workspace_test/strongbox/strongbox-web-core/src/test/java/org/carlspring/strongbox/config/IntegrationTest.java uses or overrides a deprecated API.
```
Please let me know if you want it in a different format and what if I should add something to this PR.

v3rtumnus commented:
Can be merged from my side, as the internal deprecations will be dealt with in a separate PR

fuss86 commented on strongbox-data-service/src/main/java/org/carlspring/strongbox/data/server/InMemoryOrientDbServer.java:
```diff
@@ -62,20 +62,6 @@ public void onDrop(ODatabaseInternal iDatabase)
-    public void onCreateClass(ODatabaseInternal iDatabase,
-                              OClass iClass)
-    {
-        iDatabase.registerHook(genericEntityHook);
```
@sbespalov what do you think about this change ?

sbespalov replied:
seems that this methods was deprecated
anyway I think this shouldn't affect for us, so let it be so

sbespalov commented on strongbox-commons/src/main/java/org/carlspring/strongbox/util/ClassLoaderFactory.java:
```diff
@@ -40,7 +40,7 @@ public static URLClassLoader urlClassLoaderFromPaths(ClassLoader parent,
         catch (IOException e)
         {
-            throw Throwables.propagate(e);
+            throw new RuntimeException(e);
```
I think that better to use `UndeclaredThrowableException` here and in similar cases

v3rtumnus commented:
I incorporated the feedback by @sbespalov regarding exception handling and added one missed deprecation handling for mockito. Can by merged imho when the Jenkins server is back up.

carlspring commented:
Thanks for the reviews and fixes! :)

steve-todorov commented:
@carlspring I believe this can be merged now. Failing test cases are related to another issue. 
@v3rtumnus thanks for your contribution! :+1:  :)

carlspring commented:
Oh, i didn't see the Sonar warnings before merging. @v3rtumnus : Would you mind cleaning up the warnings about the unused import statements and opening a new pull for this? Thanks!
